### ISSUE TITLE
[Suggestion] Auto-open next email or a thread on archive, delete or move. On thread selection display top mail in it.

### ISSUE THREAD

**Issue by lainverse** [bkaankose/Wino-Mail#49]
It's somewhat inconvenient to select every mail manually when you look through latest emails and archive/delete stuff of low importance. Why not auto-open them like Windows Mail does? Besides, Wino already move focus to them. Just doesn't open.

So, when active mail is archived, deleted, moved or disappeared from the list for any other reason move focus to the one directly below it and open it. When next element is a thread — select the entire thread and auto-open top email in it.

BTW, in Windows Mail I could select threads which displayed top email in them (without expanding them) and I could perform actions on the entire thread like archive or delete all mails in it. Here I can only select a thread, but it expands automatically and I have to manually select emails in it. And I can't archive or delete the entire thread as I understand. Changing threads behavior to how Windows Mail works should make navigation with keyboard more comfortable too since selecting a thread will actually display something useful from it rather than nothing and won't bloat the list unless user specifically click on expand button or use some key to expand the thread (space or right, perhaps?).

**Comment by bkaankose** [comment]
Thanks for the feedback. I agree with all your points. There's just technical challenges with managing threads at the moment. That is the main priority I postpone this enhancements. I will provide more comfortable code to play around with those threads and improve the UX.

Next version will have auto-selection of the next item under this settings menu though.

![image](https://github.com/bkaankose/Wino-Mail/assets/12009960/d4c3cadf-8b6f-4dba-9c79-4b9030b4bc71)

One of the mail goals of Wino is to provide the same experience from Windows Mail, and all these enhancements will make up to that goal eventually.

**Comment by bkaankose** [comment]
This behavior is implemented for the next version 1.7.0. Labeling as fixed for now.

Thank you for the feedback.

**Comment by bkaankose** [comment]
Version 1.7.2 is released today to Microsoft Store. Auto-selecting next item when the selected item goes out of the list is implemented, with an option to disable it under Settings > Reading Pane

![image](https://github.com/bkaankose/Wino-Mail/assets/12009960/b16879df-0937-4c47-952f-c320eacc7ade)

Closing as fixed.

**Comment by lainverse** [comment]
Well, I haven't tried Wino Mail in a while and only now tried this feature:
* Deleting an email with a hotkey moves focus out of the messages list, so you can't keep navigating it without pressing Shift+Tab a few times or clicking manually.
* When focus moved to a thread (by hotkeys or this feature) it doesn't display top email in a thread.
* The only case when the top email in a thread is displayed is when you click on a thread first, select top email, then collapse it and re-select with hotkeys or this feature by deleting an email above.

So, this feature has issues, and threads are unusable via keyboard navigation, as they were back in 2023.
Should I open a new issue about this? Were there any changes in 2.0 and should I just wait until it releases?

**Comment by bkaankose** [comment]
Beta has reworked mail list which may have address some of those issues. I am not much of a keyboard navigator so don't know the patterns that much, but you may give it a try. I'll try to improve it for upcoming releases.

**Comment by lainverse** [comment]
I tried 2.0.2.0 yesterday, and it failed to start, but 2.0.3.0 seems fine.
Behavior in 2.0 indeed changed, but for better or for worse, I'm not sure. There are some improvements and new inconveniences.

* Auto-select next on delete seems to properly select the next email and auto-open it, as expected.
* Keyboard navigation via up/down no longer auto-opens an email under the cursor (now requires pressing Enter/Space). This is not expected and is different from the most email clients I've tried.  It might be a good idea to make this configurable.
* Is there a hotkey to return to the email list? Usually, email clients return it there automatically whenever up/down is pressed outside of a context that makes use of them (like other lists or email reader/composer). At the bare minimum, it should be returned when elements above the list (select all, refresh, etc) or email action buttons on the right panel (sender, reply, delete, etc.) are in focus and up/down is pressed.
* Confusing behavior: The delete action removes the currently open email instead of the email under the cursor. In a way, linked to the above.
* Improvement: Navigating to a group does not open the topmost email, same as before, but now it's possible to expand/collapse it via Enter or Space, and it opens the top-most email.
*  However, a group has to be expanded to read or respond to the email. I don't remember how groups were handled in the old Windows Mail (only the topmost email that could be expanded as a group?), but Gmail and Mailspring don't bother displaying groups in the list at all. Only the topmost email appears in the list, with an indicator of a group and the number of messages in it. The chain of emails is displayed in the reader section. I'm not requesting this behavior, but expanding a group is inconvenient when all you care about is the topmost email in the chain. Especially when you deal with long chains.
* Additionally, this grouping behavior causes a group to jump to the top of the list the moment you press "Reply", because a temporary email is created in the group.
* It's possible to navigate emails inside a group via keyboard, but I had to press Tab multiple times before focus landed on the inside. It's very inconvenient.

IMHO, the group's root in the list should be the topmost email in the chain, and there should be a separate hotkey to expand/collapse it. So, when selected, it won't expand the group, but will open the topmost email. When expanded, focus should seamlessly move in and out of the group, like it's part of the same list, and the group borders are only visually there. I think this should be closer to old Windows Mail and won't require completely changing how the groups work.
